ext/opentelemetry-ext-aiopg/src/opentelemetry/ext/aiopg/aiopg_integration.py

```diff
@@ -0,0 +1,144 @@
+import typing
+
+import wrapt
+from aiopg.utils import _ContextManager, _PoolAcquireContextManager
+
+from opentelemetry.ext.dbapi import DatabaseApiIntegration, TracedCursor
+from opentelemetry.trace import SpanKind
+from opentelemetry.trace.status import Status, StatusCanonicalCode
+
+
+# pylint: disable=abstract-method
+class AsyncProxyObject(wrapt.ObjectProxy):
+    def __aiter__(self):
+        return self.__wrapped__.__aiter__()
+
+    async def __anext__(self):
+        result = await self.__wrapped__.__anext__()
+        return result
+
+    async def __aenter__(self):
+        return await self.__wrapped__.__aenter__()
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        return await self.__wrapped__.__aexit__(exc_type, exc_val, exc_tb)
+
+    def __await__(self):
+        return self.__wrapped__.__await__()
+
+
+class AiopgIntegration(DatabaseApiIntegration):
+    async def wrapped_connection(
+        self,
+        connect_method: typing.Callable[..., typing.Any],
+        args: typing.Tuple[typing.Any, typing.Any],
+        kwargs: typing.Dict[typing.Any, typing.Any],
+    ):
+        """Add object proxy to connection object.
+        """
+        connection = await connect_method(*args, **kwargs)
+        # pylint: disable=protected-access
+        self.get_connection_attributes(connection._conn)
+        return get_traced_connection_proxy(connection, self)
+
+    async def wrapped_pool(self, create_pool_method, args, kwargs):
+        pool = await create_pool_method(*args, **kwargs)
+        async with pool.acquire() as connection:
+            # pylint: disable=protected-access
+            self.get_connection_attributes(connection._conn)
+        return get_traced_pool_proxy(pool, self)
+
+
+def get_traced_connection_proxy(
+    connection, db_api_integration, *args, **kwargs
+):
+    # pylint: disable=abstract-method
+    class TracedConnectionProxy(AsyncProxyObject):
+        # pylint: disable=unused-argument
+        def __init__(self, connection, *args, **kwargs):
+            super().__init__(connection)
+
+        def cursor(self, *args, **kwargs):
+            coro = self._cursor(*args, **kwargs)
+            return _ContextManager(coro)
```

Comment:
Can we just call `self.__wrapped__.cursor` instead?

Reply:
There is required to wrap 'Cursor' by `get_traced_connection_proxy`.
`cursor` is syncronous function. It receive coroutine when `_cursor` is called, which can not be awaited to get Cursor. That is why it is wrapped by `get_traced_connection_proxy` in `_cursor` function. `_cursor` is async, and it is possible to await `self.__wrapped__._cursor` to get `Cursor` to wrap it.